src/display/graphics.c

```diff
@@ -0,0 +1,387 @@
+	mutex_take(_display_mutex, TIMEOUT_MAX);
+	vexDisplayCircleDraw(x, y, r);
+	mutex_give(_display_mutex);
+}
+void display_clear_circle(int16_t x, int16_t y, int16_t r) {
+	mutex_take(_display_mutex, TIMEOUT_MAX);
+	vexDisplayCircleClear(x, y, r);
+	mutex_give(_display_mutex);
+}
+void display_fill_circle(int16_t x, int16_t y, int16_t r) {
+	mutex_take(_display_mutex, TIMEOUT_MAX);
+	vexDisplayCircleFill(x, y, r);
+	mutex_give(_display_mutex);
+}
+
+//Text Display Functions
+void display_vprintf(uint8_t txt_fmt, const int16_t line, const char* fmt, va_list args){
+    mutex_take(_display_mutex, TIMEOUT_MAX);
+	char* out;
+	vasprintf(&out, fmt, args);
+	va_list empty;
+	switch(txt_fmt){
+		case 0:
+		case 1:
+		{
+            vexDisplayVString(line, out, empty);
+            break;
+        }
+		case 2:
+		{
+            vexDisplayVBigString(line, out, empty);
+            break;
+        }
+        case 3:
+		{
+            vexDisplayVCenteredString(line, out, empty);
+            break;
+        }
+        case 4:
+        {
+            vexDisplayVBigCenteredString(line, out, empty);
+            break;
+        }
+        default:
+        {
+            vexDisplayVString(line, out, empty);
+            break;
+        }
+    }
+	mutex_give(_display_mutex);
+}
+
+void display_vprintf_at(uint8_t txt_fmt, const int16_t x, const int16_t y, const char* fmt, va_list args){
+	mutex_take(_display_mutex, TIMEOUT_MAX);
+	char* out;
+	vasprintf(&out, fmt, args);
+	va_list empty;
+	switch(txt_fmt){
+		case 0:
+		{
+            vexDisplayVSmallStringAt(x, y, out, empty);
+            break;
+        }
+		case 1:
+		case 3:
+		{
+            vexDisplayVStringAt(x, y, out, empty);
+            break;
+        }
+		case 2:
+		case 4:
+		{
+            vexDisplayVBigStringAt(x, y, out, empty);
+            break;
+        }
+        default:
+        {
+            vexDisplayVStringAt(x, y, out, empty);
+            break;
+        }
+    }
+	mutex_give(_display_mutex);
+}
+
+void display_opacity_vprintf( int32_t xpos, int32_t ypos, uint32_t opacity, const char *format, va_list args) {
+	mutex_take(_display_mutex, TIMEOUT_MAX);
+	char* out;
+	vasprintf(&out, format, args);
+	va_list empty;
+	vexDisplayVPrintf(xpos, ypos, opacity, out, empty);
+	mutex_give(_display_mutex);
+}
+
+int16_t touch_last_x() {
+	static V5_TouchStatus v5_touch_status;
+	mutex_take(_display_mutex, TIMEOUT_MAX);
+	vexTouchDataGet(&v5_touch_status);
+	mutex_give(_display_mutex);
+	return v5_touch_status.lastXpos;
+}
+
+int16_t touch_last_y() {
+	static V5_TouchStatus v5_touch_status;
+	mutex_take(_display_mutex, TIMEOUT_MAX);
+	vexTouchDataGet(&v5_touch_status);
+	mutex_give(_display_mutex);
+	return v5_touch_status.lastYpos;
+}
+
+touch_event_e_t touch_last_event() {
+	static V5_TouchStatus v5_touch_status;
+	mutex_take(_display_mutex, TIMEOUT_MAX);
+	vexTouchDataGet(&v5_touch_status);
+	mutex_give(_display_mutex);
+	return v5_touch_status.lastEvent;
+}
```

Comment:
How big is V5_TouchStatus? Do these all need to be static? If so, can we combine these into one struct? You can get torn values otherwise.

Reply:
Don't think these need to be static, I'll make the change.